contrib/go/src/python/pants/contrib/go/tasks/go_fetch.py

```diff
@@ -93,10 +97,28 @@ def _fetch_pkg(self, gopath, pkg, rev):
     for path in os.listdir(root_dir):
       os.symlink(os.path.join(root_dir, path), os.path.join(dest_dir, path))
 
-  def _map_fetched_remote_source(self, go_remote_lib, gopath, all_known_remote_libs, resolved_remote_libs, undeclared_deps):
-    for remote_import_path in self._get_remote_import_paths(go_remote_lib.import_path, gopath=gopath):
-      fetcher = self._get_fetcher(remote_import_path)
-      remote_root = fetcher.root()
+  def _map_fetched_remote_source(self, go_remote_lib, gopath, all_known_remote_libs,
+                                 resolved_remote_libs, undeclared_deps, import_root_map):
+    # See if we've computed the remote import paths for this rev of this lib in a previous run.
+    remote_import_paths_cache = os.path.join(os.path.dirname(gopath), 'remote_import_paths.txt')
+    if os.path.exists(remote_import_paths_cache):
+      with open(remote_import_paths_cache, 'r') as fp:
+        remote_import_paths = [line.decode('utf8').strip() for line in fp.readlines()]
+    else:
+      remote_import_paths = self._get_remote_import_paths(go_remote_lib.import_path,
+                                                          gopath=gopath)
+      with safe_concurrent_creation(remote_import_paths_cache) as safe_path:
+        with open(safe_path, 'w') as fp:
+          for path in remote_import_paths:
+            fp.write('{}\n'.format(path).encode('utf8'))
+
+    for remote_import_path in remote_import_paths:
+      remote_root = import_root_map.get(remote_import_path)
+      if remote_root is None:
+        fetcher = self._get_fetcher(remote_import_path)
+        remote_root = fetcher.root()
+        import_root_map[remote_import_path] = remote_root
```

Comment:
Very much so. This is how import_root_map gets updated.

Added comments to clarify.